### ISSUE TITLE
Mining fails on Ubuntu

### ISSUE THREAD

**Issue by peter-fu** [aeternity/epoch#546]
- OS: Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-62-generic x86_64)
- Epoch version: 0.4.0

Contents from epoch_pow_cuckoo.log

```
2017-12-30 01:59:36.877 [debug] <0.2529.0>@aec_pow_cuckoo:parse_generation_result:386 Looking for 42-cycle on cuckoo28("XnRmgeyUjHL5Ipo6LP8j71ekLoxHDjfzQrdP8fq3NLI=PUXvaNfGdAY=",0) with 50% edges
2017-12-30 01:59:36.877 [error] <0.2529.0>@aec_pow_cuckoo:wait_for_result:328 OS process died: {signal,sigabrt,false}
2017-12-30 01:59:36.882 [info] <0.2531.0>@aec_pow_cuckoo:generate_int:123 Executing cmd: "env LD_LIBRARY_PATH=../lib:$LD_LIBRARY_PATH ./mean28s-generic -h XnRmgeyUjHL5Ipo6LP8j71ekLoxHDjfzQrdP8fq3NLI=PkXvaNfGdAY= -t 5"
2017-12-30 01:59:36.889 [error] <0.2531.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: terminate called after throwing an instance of '
2017-12-30 01:59:36.897 [error] <0.2531.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: std::bad_alloc
2017-12-30 01:59:36.899 [error] <0.2531.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: '

2017-12-30 01:59:36.902 [error] <0.2531.0>@aec_pow_cuckoo:wait_for_result:313 ERROR:   what():  
2017-12-30 01:59:36.904 [error] <0.2531.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: std::bad_alloc
2017-12-30 01:59:36.906 [error] <0.2531.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: 

2017-12-30 01:59:36.911 [debug] <0.2531.0>@aec_pow_cuckoo:parse_generation_result:386 Looking for 42-cycle on cuckoo28("XnRmgeyUjHL5Ipo6LP8j71ekLoxHDjfzQrdP8fq3NLI=PkXvaNfGdAY=",0) with 50% edges
2017-12-30 01:59:36.911 [error] <0.2531.0>@aec_pow_cuckoo:wait_for_result:328 OS process died: {signal,sigabrt,false}
2017-12-30 01:59:36.915 [info] <0.2533.0>@aec_pow_cuckoo:generate_int:123 Executing cmd: "env LD_LIBRARY_PATH=../lib:$LD_LIBRARY_PATH ./mean28s-generic -h XnRmgeyUjHL5Ipo6LP8j71ekLoxHDjfzQrdP8fq3NLI=P0XvaNfGdAY= -t 5"
2017-12-30 01:59:36.924 [error] <0.2533.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: terminate called after throwing an instance of '
2017-12-30 01:59:36.926 [error] <0.2533.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: std::bad_alloc
2017-12-30 01:59:36.928 [error] <0.2533.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: '

2017-12-30 01:59:36.932 [error] <0.2533.0>@aec_pow_cuckoo:wait_for_result:313 ERROR:   what():  
2017-12-30 01:59:36.939 [error] <0.2533.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: std::bad_alloc
2017-12-30 01:59:36.941 [error] <0.2533.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: 

2017-12-30 01:59:36.943 [debug] <0.2533.0>@aec_pow_cuckoo:parse_generation_result:386 Looking for 42-cycle on cuckoo28("XnRmgeyUjHL5Ipo6LP8j71ekLoxHDjfzQrdP8fq3NLI=P0XvaNfGdAY=",0) with 50% edges
2017-12-30 01:59:36.943 [error] <0.2533.0>@aec_pow_cuckoo:wait_for_result:328 OS process died: {signal,sigabrt,false}
2017-12-30 01:59:36.948 [info] <0.2535.0>@aec_pow_cuckoo:generate_int:123 Executing cmd: "env LD_LIBRARY_PATH=../lib:$LD_LIBRARY_PATH ./mean28s-generic -h XnRmgeyUjHL5Ipo6LP8j71ekLoxHDjfzQrdP8fq3NLI=QEXvaNfGdAY= -t 5"
2017-12-30 01:59:36.957 [error] <0.2535.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: terminate called after throwing an instance of '
2017-12-30 01:59:36.959 [error] <0.2535.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: std::bad_alloc
2017-12-30 01:59:36.961 [error] <0.2535.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: '

2017-12-30 01:59:36.964 [error] <0.2535.0>@aec_pow_cuckoo:wait_for_result:313 ERROR:   what():  
2017-12-30 01:59:36.966 [error] <0.2535.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: std::bad_alloc
2017-12-30 01:59:36.968 [error] <0.2535.0>@aec_pow_cuckoo:wait_for_result:313 ERROR: 
```

**Comment by lucafavatella** [comment]
@peter-fu Thanks for your report. It looks like your node has memory allocation issues, probably because of memory constraints.

What is your amount of memory (RAM)? Do you use swap memory? The current recommended amount of memory is 4 GB (see release notes).

I believe that the issue you experienced is addressed in [release 0.4.1](https://github.com/aeternity/epoch/releases/tag/v0.4.1), that introduced the possibility for the user to configure a less memory-intensive (though usually slower) mining algorithm. Its release notes include  the instructions ([“Troubleshooting node failing mining attempts”](https://github.com/aeternity/epoch/blob/8e59d7a2d19811be7c47377274e329b7ca9501ec/RELEASE-NOTES.md#troubleshooting-node-failing-mining-attempts)). Could you please re-test and let us know?

**Comment by peter-fu** [comment]
You're right, RAM was 1GB, which was the cause of the mining failure. The 4GB recommandation was in the documentation already, I just didn't have such one around.

And I can confirm the work around works for my case.

Thanks a lot.

Contents from epoch_mining.log
```
2018-01-06 13:12:56.284 [info] <0.906.0>@aec_conductor:start_mining:643 Starting mining
2018-01-06 13:12:58.278 [info] <0.906.0>@aec_conductor:handle_synced_block:764 sync_block: {block,4225,<<135,78,166,209,208,184,146,125,162,228,1,10,111,60,223,162,50,116,17,1,254,185,162,22,90,120,109,59,47,235,132,50>>,<<90,89,43,154,76,5,195,199,211,98,61,138,160,123,163,160,150,190,219,4,223,220,99,116,84,61,70,234,23,151,159,24>>,<<152,7,90,42,104,136,64,155,145,156,143,189,128,200,89,211,250,30,95,151,113,47,171,99,199,250,183,178,194,251,203,227>>,[{signed_tx,{coinbase_tx,<<4,111,4,164,239,133,137,92,237,178,82,18,4,50,50,240,57,228,244,238,220,207,28,61,11,217,153,178,185,77,243,138,88,160,35,231,106,18,248,5,71,6,8,157,155,236,147,55,155,67,124,67,128,131,32,93,120,164,135,200,98,230,175,231,14>>},[<<48,69,2,33,0,136,148,66,164,138,54,137,231,168,198,143,144,6,245,114,69,97,117,209,160,52,215,42,14,252,121,96,48,109,236,18,206,2,32,33,197,212,16,50,106,253,92,118,14,92,58,91,106,251,164,245,64,50,234,115,78,10,207,18,40,67,150,98,89,139,50>>]}],537418489,8735000823384452916,1515215338618,3,[2882867,3494520,6563495,7215916,7822962,9190290,10495184,13088799,14219175,16416902,16798941,20719297,21895187,24749427,26826880,28115045,30165697,37982250,39696925,45957400,52474284,53160333,58427210,73148109,80029939,87510353,87674783,97596014,99595439,103683000,105301757,109727499,112293103,114104660,118534764,124972736,125360634,125727056,125924421,127846268,131714047,133029493]}
2018-01-06 13:12:58.292 [info] <0.906.0>@aec_conductor:update_tx_pool_on_top_change:587 Updating transactions <<135,78,166,209,208,184,146,125,162,228,1,10,111,60,223,162,50,116,17,1,254,185,162,22,90,120,109,59,47,235,132,50>> <<97,3,74,204,193,1,142,64,93,240,115,53,67,20,183,39,122,251,209,214,53,98,114,146,171,217,209,202,201,11,122,142>>
2018-01-06 13:12:58.292 [info] <0.906.0>@aec_conductor:create_block_candidate:728 Creating block candidate
2018-01-06 13:12:58.329 [info] <0.906.0>@aec_conductor:handle_block_candidate_reply:744 Created block candidate and nonce (max 17260033612426419466, current 17260033612426419467).
2018-01-06 13:12:58.329 [info] <0.906.0>@aec_conductor:start_mining:643 Starting mining
```
